avocado/utils/process.py

```diff
@@ -199,22 +200,41 @@ def get_children_pids(parent_pid, recursive=False):
     return children
 
 
-def kill_process_tree(pid, sig=signal.SIGKILL, send_sigcont=True):
+def kill_process_tree(pid, sig=signal.SIGKILL, send_sigcont=True,
+                      wait=False):
     """
     Signal a process and all of its children.
 
     If the process does not exist -- return.
 
     :param pid: The pid of the process to signal.
     :param sig: The signal to send to the processes.
+    :param wait: How long to wait for the pid(s) to die (False=don't wait,
+                 None=wait for infinity, otherwise wait for $wait seconds)
     """
+    if not wait:
+        def remaining_time():
+            return wait
+    else:
+        start = time.time()
+
+        def remaining_time():
+            return wait - time.time() + start
+
     if not safe_kill(pid, signal.SIGSTOP):
         return
     for child in get_children_pids(pid):
-        kill_process_tree(int(child), sig)
+        kill_process_tree(int(child), sig, wait=remaining_time())
     safe_kill(pid, sig)
     if send_sigcont:
         safe_kill(pid, signal.SIGCONT)
+    if wait is False:
+        return
+    elif wait is not None:
+        wait_for(lambda: not pid_exists(pid), remaining_time())
+    else:
+        while pid_exists(pid):
+            time.sleep(0.1)
```

Comment:
I believe it makes sense to make the sleep between cycles either a parameter or a documented characteristic, because this may conflict with very short `wait` times given as a parameter.  What I mean is, in theory, a user that gives `wait=0.05` would expect the wait to not be too much longer than that as a whole.

Reply:
Well I wanted to start with something simple and well behave and if users need that they can send PRs. Do you think it's worth the extra parameter? (I mean for years we lived without sleep at all)